binaries/build.py

```diff
@@ -0,0 +1,127 @@
+import sys
+import shutil
+import os
+import yaml
+
+# os.chmod() is called for every binary at import time,
+# before we put them in wheels.
+import streamer_binaries
+
+
+class custom_build_py(build_py):
+  """A custom class to override the default behavior of `build_py` command."""
+
+  def run(self):
+    # Clean the build directory so the binaries that were added at the previous
+    # build don't remain in the package for the current platform we build for.
+    # This is because setuptools doesn't delete it after a build has
+    # completed and doesn't also re-create it in the next `build` call,
+    # that's why we can't delete it either, so let's just clean it.
+    build_dir = os.path.join(self.build_lib, streamer_binaries.__name__)
+    self._clean_build_dir(build_dir)
+    return super().run()
+
+  def _clean_build_dir(self, build_dir):
+    try:
+      # To clean the directory, we can remove it and re-create it.
+      shutil.rmtree(build_dir)
+      os.mkdir(build_dir)
+    except FileNotFoundError:
+      # At the first run, the directory might not be on the system yet.
+      return
+
+
+def build_wheel(platform_name, platform_binaries):
+  """Builds a wheel distribution for `platform_name` adding the files
+  in `platform_binaries` to it using the `package_data` parameter."""
+
+  sys.argv = [
+      'setup.py',
+      # Build binary as a wheel.
+      'bdist_wheel',
+      # Platform name to embed in generated filenames.
+      '--plat-name', platform_name,
+      # Temporary directory for creating the distribution.
+      '--bdist-dir', platform_name,
+      # Python tag to embed in the generated filenames.
+      '--python-tag', 'py3',
+      # Run quietly.
+      '--quiet',
+  ]
+
+  # This setup() call will ingest the sys.argv command line arguments.
+  setuptools.setup(
+    name='shaka-streamer-binaries',
+    version=streamer_binaries.__version__,
+    author='Google',
+    description='A package containing FFmpeg, FFprobe, and Shaka Packager static builds.',
+    long_description=('An auxiliary package that provides platform-specific'
+                      ' binaries used by Shaka Streamer.'),
+    url='https://github.com/google/shaka-streamer/tree/master/binaries',
+    packages=[streamer_binaries.__name__,],
+    classifiers=[
+        'Programming Language :: Python :: 3',
+        'License :: OSI Approved :: Apache Software License',
+        'Operating System :: POSIX :: Linux',
+        'Operating System :: MacOS :: MacOS X',
+        'Operating System :: Microsoft :: Windows',
+    ],
+    package_data={
+        # Only add the corresponding platform specific binaries
+        # to the package for the current platform_name.
+        streamer_binaries.__name__: platform_binaries,
+    },
+    # Use our custom builder.  All it does is cleaning the build directory
+    # before using it for building, as it might contain old unwanted binaries.
+    cmdclass={'build_py': custom_build_py},
```

Comment:
Could you consider using [`TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) instead?  Something like:

```py
with tempfile.TemporaryDirectory() as temp_dir_name:
  download_into(temp_dir_name, platform_name)
  build_binary_package(temp_dir_name, platform_name)
# Now the temporary directory and its contents have been wiped out
```

Reply:
If i get this correctly, you mean to download the binaries in a temp directory and then add it to the package.
`setup` looks for `package_data` inside the package we say it resides in(in this case, inside `streamer_binaries`), i don't think we can put it some where else. If `setup` couldn't find the files we listed in `package_data` inside the package, it just skips it, doesn't search somewhere else.

Reply:
I looked up a bit about whether we are able to specify the build directory to the `bdist_wheel` command, but found we no way.  `bdist_wheel` calls `build` internally, while the `build` command has the `--build-lib` option exposed, the `bdist_wheel` doesn't.
Maybe the usage of `cmdclass` is a bit confusing.  We can simplify this to be only creating the needed build directory before the `setup` call and removing it afterwards, after the `setup` call.

Reply:
I did change it a little bit now to use the temporary directory `--bdist-dir`. This means we build directly in the same directory we package things in afterwards, `setuptools` removes this directory after building the wheel.
Kinda hacky, but i think is simpler.